### DISCUSSION TITLE
Storage::putFile strips leading slash

### DISCUSSION THREAD

**Discussion by Lombra** [laravel/framework#47291] [Q&A]
Hi,

Trying to upload and store some files using `Storage::putFile('/images')` and storing the resulting path in the database, which then gets used directly as hrefs. `putFile` strips the leading slash however, resulting in a different URL entirely on most pages. Am I misunderstanding what the path argument is for, or why would it strip the leading slash? Trailing slash I might understand.

**Comment by Noorani-MM** [comment]
Hi.

This kind of url contains 3 section. 1. Base url (server), 2. directory url, 3. File name

Usually you have to store File name and by type model you have to realize the store directory and base url come from `env` file

So you have to do something like this

```php
$base_url = env('APP_URL');

$url = "{$base_url}/{$directory_url}/{$model->href}";
```

Now you have a more semantically understandable and decoupled code
